pkg/api/handlers/compat/auth.go

```diff
@@ -41,18 +36,26 @@ func Auth(w http.ResponseWriter, r *http.Request) {
 	sysCtx := runtime.SystemContext()
 	sysCtx.DockerInsecureSkipTLSVerify = skipTLS
 
-	fmt.Println("Authenticating with existing credentials...")
-	registry := stripAddressOfScheme(authConfig.ServerAddress)
-	if err := DockerClient.CheckAuth(r.Context(), sysCtx, authConfig.Username, authConfig.Password, registry); err == nil {
+	loginOpts := &auth.LoginOptions{
+		Username:    authConfig.Username,
+		Password:    authConfig.Password,
+		Stdout:      io.Discard,
+		NoWriteBack: true, // to prevent credentials to be written on disk
+	}
+	if err := auth.Login(r.Context(), sysCtx, loginOpts, []string{authConfig.ServerAddress}); err == nil {
 		utils.WriteResponse(w, http.StatusOK, entities.AuthReport{
 			IdentityToken: "",
 			Status:        "Login Succeeded",
 		})
 	} else {
 		var msg string
 
+		// Return a 401 in case the credentials are wrong.  The string
+		// compare is needed because `auth.Login` turns
+		// `ErrUnauthorizedForCredentials` into something friendly to
+		// the human eyes.
 		var unauthErr DockerClient.ErrUnauthorizedForCredentials
-		if errors.As(err, &unauthErr) {
+		if errors.As(err, &unauthErr) || strings.Contains(err.Error(), "logging into ") {
```

Comment:
Can we wrap the error instead? Or alternatively return a new typed error. These string matches break easily and there doesn't seem to be a test for it either.

Reply:
Ignore this comment: Looks like you are faster than me.